resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v1/LoggingPodStatusWatcher.scala

```diff
@@ -44,15 +44,18 @@ private[kubernetes] class LoggingPodStatusWatcher(podCompletedFuture: CountDownL
   private val logRunnable: Runnable = new Runnable {
     override def run() = logShortStatus()
   }
-  if (interval > 0) {
-    scheduler.scheduleWithFixedDelay(logRunnable, 0, interval, TimeUnit.MILLISECONDS)
-  }
 
   private var pod: Option[Pod] = Option.empty
   private def phase: String = pod.map(_.getStatus().getPhase()).getOrElse("unknown")
   private def status: String = pod.map(_.getStatus().getContainerStatuses().toString())
     .getOrElse("unknown")
 
+  def start(): Unit = {
+    if (interval > 0) {
+      scheduler.scheduleWithFixedDelay(logRunnable, 0, interval, TimeUnit.MILLISECONDS)
```

Comment:
+1 for using daemon threads, but IMHO it's better to explicitly start the timer so it's a bit easier to reason about the control flow.